runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContext.java

```diff
@@ -361,7 +387,9 @@ private void recordActiveMessageInProcessingTimesMap() {
             if (v == null) {
               v = new IntSummaryStatistics();
             }
-            v.accept((int) (System.currentTimeMillis() - this.activeMessageMetadata.startTime()));
+            synchronized (this) {
```

Comment:
```suggestion
```
This is redundant since compute is synchronous and the callback is called synchronously.

Reply:
The line still shows up in the diff, need to remove.

Reply:
Ping on this, I agree with Arun should already be handled by enclosed synchronized block